packages/manager/apps/pci-gateway/src/pages/list/ListGuard.tsx

```diff
@@ -0,0 +1,31 @@
+import { useEffect, useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+import { useAllAggregatedGateway } from '@/api/hooks/useGateway';
+
+export default function ListGuard({
+  projectId,
+  children,
+}: {
+  projectId: string;
+  children: JSX.Element;
+}): JSX.Element {
+  const [displayChildren, setDisplayChildren] = useState(false);
+
+  const navigate = useNavigate();
+
+  const { data: aggregatedGateways, isLoading } = useAllAggregatedGateway(
```

Comment:
isPending instead of isLoading, specially if useAllAggregatedGateway depend to another fetch, the isLoading will be false in first render